python/ray/util/client/worker.py

```diff
@@ -320,6 +334,27 @@ def _call_schedule_for_task(
                 logger.exception("Failed to deserialize {}".format(
                     ticket.error))
                 raise
+        self.num_tasks_scheduled += 1
+        self.total_outbound_message_size += task.ByteSize()
+        if not self.warning_raised and \
+                self.num_tasks_scheduled > TASK_WARNING_THRESHOLD:
+            logger.warning(
+                f"More than {TASK_WARNING_THRESHOLD} remote tasks have been "
+                "scheduled. This can be slow on Ray Client due to "
+                "communication overhead. If you're running many fine-grained "
+                "tasks consider batching them (details in the Ray Design "
```

Comment:
Ray Design Pattern? where is that?
CC @amogkam

Reply:
https://docs.google.com/document/d/167rnnDFIVRhHhK4mznEIemOtj63IOhtIPvSYaPgI4Fg/edit

It's currently linked in the repo's main repo near the bottom next to the architecture/internals docs, but could also either link it directly in the error message or just suggest batching without mentioning it.

Reply:
Maybe we should ask the user to put everything in a remote function?
I am not sure if "batching them" is clear enough. Maybe we should say put them in a remote function.